### PR TITLE
Fix for saving designer page with already existing names

### LINKED ISSUE #15432: Names of the pages in the designer could be saved with a already existing name
### Describe the bug

Pages in the designer can be saved with the exiting names.

### To Reproduce

Steps to reproduce the behaviour:
1. Go to designer
2. Click on 'Save page' in the left panel
3. Try to save a page with some name. (let's say qwe here)
4. Save multiple pages using the same name(qwe here).

### Expected behaviour

It should throw and error saying "there already exists a page named 'qwe', please rename it to something else".

### Screenshots

Complete screen:

Selection_073

When clicked on the 'select page' drop down:

Menu_070

### PR DESCRIPTION
**Pull request by tabraiz961**
### Fix for saving designer page with already existing names 
Fixes #15432

### CODE REVIEW

**Comment by tabraiz961** [comment]
@williamdes Did i miss something cause i am seeing this error

> No coverage uploaded for pull request base (`QA_5_0@d01dab8`)

**Comment by williamdes** [comment]
@tabraiz961 nevermind, this is not important in your case
I will review the PR as soon as possible

**Comment by tabraiz961** [comment]
hi @williamdes  is the Codacy/PR Quality Review something to worry about? I have provided Codacy permission from  my github settings still it is unable to review the pull request.

File: db_designer.php [hunk 1]
```diff
@@ -76,8 +76,17 @@
         if ($_POST['save_page'] == 'same') {
             $page = $_POST['selected_page'];
         } else { // new
-            $page = $designerCommon->createNewPage($_POST['selected_value'], $_POST['db']);
-            $response->addJSON('id', $page);
+            if ($designerCommon->getPageExists($_POST['selected_value'])!=-1) {
+                $response->addJSON(
+                    'message',
+                    "There already exists a page named '" . $_POST['selected_value'] . "', please rename it to something else."
```
Comment (williamdes) [inline db_designer.php]:
```suggestion
                    _(sprintf("There already exists a page named "%s" please rename it to something else.", htmlentities($_POST['selected_value'])));
```

Comment (williamdes) [reply db_designer.php]: That will make it translatable but a security check must be done

File: libraries/classes/Database/Designer/Common.php [hunk 2]
```diff
@@ -390,6 +390,38 @@ public function getDefaultPage($db): ?int
...
 
+    /**
+     * Get the status if the page already exists
+     * If no such exists, returns negative index.
+     *
+     * @param string $pg name
+     *
+     * @return int status of already existing page
+     */
+    public function getPageExists($pg)
```
Comment (williamdes) [inline libraries/classes/Database/Designer/Common.php]:
```suggestion
    public function getPageExists(string $pg): int
```

**Comment by williamdes** [comment]
No problem for codacy, you can ignore it
But be sure travis CI passes (except Google Chrome tests)

File: db_designer.php (hunk 1 above)
Comment (williamdes) [reply db_designer.php]: Also you can add a line `/* l10n: some explanation for translators here */`
You can find examples in our codebase

Comment (tabraiz961) [reply db_designer.php]: > l10n

I could not find explanation for translator (other than variable declarations) from others in my specific section thats why didn't provide some! however i'd be happy to help if you provide some suggestions.

**Comment by williamdes** [comment]
@tabraiz961 please do `git reset - - hard 9f892b561d6b55d6690510038cf52f1f01645679`
And `git push - - force`
That is an inconvenience when using the main QA_5_0 branch for a PR
I will try to merge it soon 
Thank you

**Comment by tabraiz961** [comment]
Hi @williamdes  thanks 

> @tabraiz961 please do `git reset --hard 9f892b561d6b55d6690510038cf52f1f01645679`
> And `git push --force`
> That is an inconvenience when using the main QA_5_0 branch for a PR
> I will try to merge it soon
> Thank you
Thanks so much i was scratching my head here keep up the good work

**Comment by williamdes** [comment]
Hi @tabraiz961 did you succeed to fix it ?
I can do it on my end if you want, I prefer that you do it so you are able to fix it next time ;)

**Comment by tabraiz961** [comment]
> Hi @tabraiz961 did you succeed to fix it ?
> I can do it on my end if you want, I prefer that you do it so you are able to fix it next time ;)

I changed my email once after commiting, i am reverting it. That was causing the signoff failure :(

**Comment by williamdes** [comment]
@tabraiz961 I have forced DCO to pass since your email is not an issue.

**Review by williamdes — APPROVED** [review]

**Comment by williamdes** [comment]
Waiting for CI tests and lint to pass

**Comment by williamdes** [comment]
Merged, thank you for your contribution !

**Pull request merged**